pkg/actuators/machineset/controller.go

```diff
@@ -121,7 +121,8 @@ func reconcile(machineSet *machinev1.MachineSet) (ctrl.Result, error) {
 
 	// TODO: get annotations keys from machine API
 	machineSet.Annotations[cpuKey] = strconv.FormatInt(instanceType.VCPU, 10)
-	machineSet.Annotations[memoryKey] = strconv.FormatInt(instanceType.MemoryMb, 10)
+	// See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory
+	machineSet.Annotations[memoryKey] = fmt.Sprintf("%sM", strconv.FormatInt(instanceType.MemoryMb, 10))
```

Comment:
Need to work out what value the provider is returning to determine if `M` or `Mi` should be used

Reply:
yeah, i think it's worth digging a little more. "M" seemed the safest value from the docs i read on the kube side of things.

Reply:
In the autoscaler (and pretty much everywhere else) I think Mi is used, so using M here will change the current behaviour. I suspect this should be Mi, but would like to see the docs to confirm

Reply:
seems like the values are coming directly from this table, https://github.com/openshift/cluster-api-provider-aws/blob/master/pkg/actuators/machineset/ec2_instance_types.go#L29

which makes me think "M" is probably want we want here.

Reply:
The AWS docs for instances (https://aws.amazon.com/ec2/instance-types/) list the values as `GiB`, so an `a1.2xLarge` they claim is `16Gi`, the autoscaler table you linked has it as 16384, so I think that is in `Mi` rather than `M`

Reply:
ok, after looking at this from aws[0] and this from wikipedia[1], i suspect those are MiB values in our table. i'll adjust this.

[0] https://aws.amazon.com/ec2/instance-types/
[1] https://en.wikipedia.org/wiki/Mebibyte